Implement visual regression tests for error pages

MKodde opened the pull request:
Different error pages are visited and compared to previous snapshots
when a screenshot differs from a previous one, the test will fail. This
test should be run manually for now.

See task 3 of https://www.pivotaltracker.com/story/show/165593518

pablothedude commented:
Great work, this is going to be very valuable to test for regressions later on, maybe consider to enable this after we have finished the 5.11 release?

1. In the initial run I got a timeout... maybe the timeout could be set less strict.
1. In the second run the diff doesn't seem to work because the loop over the viewports in the same test and a race condition in the viewports diffs the image with the wrong size so it fails. I would suggest a test per viewport and error page  and not only per error page or a custom identifier to keep the names a bit shorter. This in order to quickly see which dimension is broken even if we support more dimensions.
1. Also I would like to have a fullage snapshot to also compare the hidden areas for small viewports.

MKodde commented:
Thanks, 

1. Upped the timeouts, hopefully, this resolves your issues
2. Fixed
3. Fixed, thanks for the fullpage tip.

As a bonus I looked into also adding locale support (rendering all pages in NL and EN) but could not manipulate the lang cookie. Might be a rookie mistake on my end, but postponing that for now.
